cluster-autoscaler/utils/gpu/gpu.go

```diff
@@ -156,3 +158,36 @@ func GetGpuRequests(pods []*apiv1.Pod) map[string]GpuRequestInfo {
 	}
 	return result
 }
+
+// GetNodeTargetGpus returns the number of gpus on a given node. This includes gpus which are not yet
+// ready to use and visible in kubernetes.
+func GetNodeTargetGpus(node *apiv1.Node, nodeGroup cloudprovider.NodeGroup) (gpuType string, gpuCount int64, err error) {
+	gpuLabel, found := node.Labels[GPULabel]
+	if !found {
+		return "", 0, nil
+	}
+
+	gpuAllocatable, found := node.Status.Allocatable[ResourceNvidiaGPU]
+	if found && gpuAllocatable.Value() > 0 {
+		return gpuLabel, gpuAllocatable.Value(), nil
+	}
+
+	// A node is supposed to have GPUs (based on label), but they're not available yet
+	// (driver haven't installed yet?).
+	// Unfortunately we can't deduce how many GPUs it will actually have from labels (just
+	// that it will have some).
+	// Ready for some evil hacks? Well, you won't be disappointed - let's pretend we haven't
+	// seen the node and just use the template we use for scale from 0. It'll be our little
+	// secret.
```

Comment:
I appreciate you leaving my original comment here :+1:

Reply:
I left whole method. Why would I remove such a beatiful comment :)